docs/concepts/time-and-windows.rst

```diff
@@ -0,0 +1,118 @@
+.. _time-and-windows:
+
+Time and Windows
+################
+
+During the life cycle of a record, it experiences a number of events
+which occur at particular times. Three important times in a record's 
+life cycle are *Event-time*, *Ingestion-time*, and *Processing-time*. 
+
+Representing time consistently enables aggregation operations on streams,
+like SUM, that have time boundaries.
+
+KSQL supports the following notions of time:
+
+Event-time

```

Comment:
> 'Event-time' is the time at which the event occurred.

That is semantically correct. Note, that we assume that users would set this timestamp in ProducerRecrod explicitly.

> It's possible the data-source, or the producing app, or what ever, is producing a record to Kafka for an event that happened a some point in the past. The 'event time' in this context would be the time the event occurred, i.e. the historical point in time.

Agreed. Again, the producer application would need to set the timestamp accordingly when creating ProducerRecords.

Bottom line: It think it makes sense to discuss semantic on producer application side and KSQL side separately.

(1) For producer side, we need to explain how event-time semantics can be achieved. Btw: often "the time at which the event occurred" is the same as "the wall-clock time when `ProducerRecord` is created" (ie, for any live/online application) -- only if you write historical data, both would be different.

(2) For KSQL, the assumption is always event-time. If the producer does it differently, the end-to-end pipeline might be incorrect (but it might also be ok, depending on the application). However, this part is not a KSQL concern any longer IMHO.